AK/SourceLocation.h

```diff
@@ -25,6 +25,13 @@ class SourceLocation {
     }
 
     constexpr SourceLocation() = default;
+    constexpr SourceLocation(SourceLocation const&) = default;
+
+    SourceLocation& operator=(SourceLocation& other)
+    {
+        new (this) SourceLocation(other.m_file, other.m_line, other.m_function);
+        return *this;
+    }
```

Comment:
wut

Can we write this as:
```cpp
constexpr SourceLocation& operator=(SourceLocation&) = default;
```

I guess we might have to make the members non-const but that seems okay(?)